### ISSUE TITLE
3U DAC (Zotino) tests & rev 1 errata

### ISSUE THREAD

**Issue by gkasprow** [m-labs/sinara#227]
Issues:
- [x] Use two output resistors in series/parallel to improve power handling (200mW short-circuit)
- [x] reverse C143, C145, C147, C84
- [x] mirrored SOT23 diode footprint due to Altium bug - part of component is on top layer, pads are on bottom layer.
- [ ] add Shottky diodes at the LDO outputs
- [ ] 3mm higher EMC shield on PSU
- [x] reverse IDC connector on BNC to IDC board
- [ ] Zero-index all channels
- [x] swap BNC connector GND terminals
- [ ] either add ~2mm plastic washers on BNC barrel **behind** front panel or move BNC connectors 2mm towards FP. This is to enable fixing the BNC connectors to the panel using the screws.
- [x]  Please remove the annotation "This module connects to Kasli ..." and replace it with something like "EEM connector: IO are LVDS, I2C is 3V3 LVCMOS, P3V3_MP up to 20mA, P12V up to 1A."
- [ ] run PI and thermal simulations
- [ ] Can we run the digital interface from 2V5?
- Temperature controller issues:
  - [x] Decrease the thermal resistance between the thermal island and the main PCB/increase the thermal mass of the island by at least 1 order of magnitude -- without active stabilization, the DAC should only be 1-2C hotter than the main PCB. This should also improve the electrical ground connection between the main PCB and the thermal island.
  -  [x] Remove current temperature controller and DAC over temperature shutdown circuitry.
  -  [x] Add mounting holes/pads for peltier + heat sink controller by external circuit.
  -  [x] Leave temperature probe on the thermal island, bring it out to an external connector, which should also provide +12V power, ground and I2C (maybe use I2C-controlled GPIO instead? This way we can do bit-banged SPI if we need to).
   - [ ] Assuming the peltier + heatsink mount on the top side of the PCB, do we want to add a stick-on insulation on the bottom side to screen the DAC from external air currents?
- [x] Silk + schematic + panel should be "v1.x" not "rev1.x"
- [x] Only have one set of +-12V regulators, not two!
- [x] Reduce SMPs outputs slightly, since we're only using +-12 regulators

Tests that can be done without an Artiq driver:
- [ ] Measure step response. This means measuring the 90%-10% rise time for a full-scale signal. If easily possible, also measure the 1% settling time.
- [ ] Measure noise PSD (0.1Hz to 10MHz) with full scale and mid scale output signal
- [ ] Glitches on LDAC, major carry, etc (trigger off the SPI)
- [ ] Confirm that temperature controller works. Measure temperature stability of DAC in lab conditions
- [ ] Measure drifts with temperature controller running. Set one channel to Vmax/0V, and look at drift over ~24 hours to confirm gain and offset drifts. Measure board temp co (e.g. use hot air gun to heat the board up while measuring output drifts).
- [ ] susceptibility to external EMI sources (i.e. neighbouring boards). I will test it with EMC equipment.

Tests that need an Artiq driver:
- [ ] Measure cross-talk (full-scale square wave at ~50kHz on one channel, another channel at 0V)
- [ ] Measure spurs due to SMPSs, digital-cross talk, pickup etc. Have one channel updating as fast as possible and measure noise on another channel
- [ ] Check for thermal issues (drifts) when outputs are slewing at max rate (e.g. testing for issues due to self-heating in the OpAmps). What’s the best way to test this?
- [ ] Measure harmonics with a full-scale sinusoid output at 50kHz.

Configuration for measurements is: DAC->3m SCSI cable->HD68 to IDC->IDC to BNC->scope/DVM/noise analyser.  CMCs and a LPF on the IDC-BNC board as previously discussed. LPF chosen to avoid degrading the DAC noise or bandwidth too much.

![3u dac photo top](https://user-images.githubusercontent.com/4325054/27735685-2d389946-5da1-11e7-87dd-faac0a5919ea.jpg)

**Comment by jordens** [comment]
Nice!
If you managed to build a switched DC-DC DAC/ADC with noise as low as envisioned, that would be a massive achievement.
I am especially looking forward to graphs/number on:

* Noise (really wideband, different output loads)
* Crosstalk
* Temperature coefficient
* Glitches on major carry

(same for #226)

**Comment by hartytp** [comment]
Great job Greg, that looks really good!

How many of these boards have arrived?

What tests are you thinking of doing (beyond confirming basic functionality)? A few things I thought about were:

- Measure step response
- Measure cross-talk (full-scale square wave at ~50kHz on one channel, another channel at 0V)
- Measure noise PSD (0.1Hz to 10MHz)
- Measure spurs due to SMPSs, digital-cross talk, pickup etc. Have one channel updating as fast as possible and measure noise on another channel
- Confirm that temperature controller works. Measure temperature stability of DAC in lab conditions
- Measure drift of channels with temperature controller running. Set one channel to Vmax/0V, and look at drift over ~24 hours to confirm gain and offset drifts.
- Check for thermal issues (drifts) when outputs are slewing at max rate (e.g. testing for issues due to self-heating in the OpAmps). What’s the best way to test this?

Configuration for measurements is: DAC->3m SCSI cable->HD68 to IDC->IDC to BNC->scope.  CMCs and a ~500kHz LPF on the IDC-BNC board as previously discussed.

**Comment by gkasprow** [comment]
I produced 5 sets of all boards.
Please make a checklist with all measurement you want me to perform.
Greg

**Comment by hartytp** [comment]
I've edited the top post with an initial list of tests to perform. Anyone should feel free to edit this if there are more tests they think we should do, or if they can think of different/better tests we should do instead.

> I produced 5 sets of all boards.

Ace. Presumably, you need 1 for testing. Could you send one to @jordens for software development, and post the other 3 to me, please? @jordens do you need a VHDCI carrier board?

**Comment by gkasprow** [comment]
Yes, I will post boards once I make them running.

**Comment by hartytp** [comment]
Thanks.

**Comment by gkasprow** [comment]
@hartytp @jordens One more thing about tests - I will do some tests that ADI evaluation software lets me perform. For rest of them I will need ARTIQ test software running on KC705 or Sayma. This approach will also validate boards operation in final environment.

**Comment by gkasprow** [comment]
DAC with shielding boxes
![2017-07-07 10 31 26](https://user-images.githubusercontent.com/4325054/27971566-744a3ea4-6353-11e7-92e9-8da6a648442a.jpg)

**Comment by hartytp** [comment]
very nice!

**Comment by gkasprow** [comment]
Status update:
Power supply works
Thermal controller needs some tuning of PI regulator 

one issue so far: DAC heats up and after a minute causes LDO overcurrent protection. Board consumes 240mA when cold and 0.4A after one minute when DAC gets hot. Needs further investigation. I will check if the issue is on other boards.

**Comment by dhslichter** [comment]
>one issue so far: DAC heats up and after a minute causes LDO overcurrent protection. Board consumes 240mA when cold and 0.4A after one minute when DAC gets hot. Needs further investigation. I will check if the issue is on other boards.

yikes!  keep us posted.

**Comment by hartytp** [comment]
@jordens @sbourdeauducq What's the status of ARTIQ support for Zotino? IIUC, @gkasprow's current plan is to test Novo using a DevKit (and some hacks to program the PGIAs), but to use Artiq to test Zotino. This will require the Artiq Zotino driver.

If possible, I'd like to have Zotino and Novo prototypes tested and shipped before Kasli/Urukul prototypes arrive to avoid building up a backlog.

**Comment by jordens** [comment]
We need the hardware to write ARTIQ support. Writing that blind is something we want to avoid. The driver for the AD53[67][0123] family is already there and tested.

Also we have the backlog from Sayma. We really need to focus on that first.

**Comment by hartytp** [comment]
> We need the hardware to write ARTIQ support. Writing that blind is something we want to avoid. 

Sorry, I thought that @gkasprow had already shipped you a Zotino and a Novo. @gkasprow Could you send M-Labs/Quartiq 1 of each of these boards, please?

> Also we have the backlog from Sayma. We really need to focus on that first.

ACK, that should take priority. But, I don't want to put off testing the Zotino prototypes for too long. So, if Artiq Zotino support is more than a few weeks off, we should look into options for testing Zotino without Artiq (e.g. use a microprocessor eval board)...

**Comment by jordens** [comment]
We may already have them in HK.
That plan is ok.
Ping @sbourdeauducq @mntng @whitequark

**Comment by gkasprow** [comment]
@hartytp I've sent them already with Saymas.

**Comment by hartytp** [comment]
@gkasprow Thanks for confirming!

@sbourdeauducq et al, let me know what you think the timeline for Artiq support of Zotino is likely to be, and we'll come up with an appropriate plan for hardware testing.

**Comment by hartytp** [comment]
@gkasprow One other thing: are there any other things that can be done on Zotino/Novo without Artiq support. e.g. IIRC, the temperature controller still needs debugging. It might be worth sorting those kinds of issue out first, while M-Labs write drivers.

**Comment by gkasprow** [comment]
Yes, I'm just preparing HDL for little CMOD7 FPGA devkit to be able to take ober the SPI bus and configure registers in ADC in Zotino and Nobo boards. I will use little FORTH CPU to do the job.
The temperature controller needs adjustment of regulator time constant - now it oscillates with sub-Hz rate.

**Comment by hartytp** [comment]
@gkasprow Thanks for the update! That sounds great. Let me know how it goes.

T

**Comment by sbourdeauducq** [comment]
We have a Zotino hooked up in HK to a KC705 using the FMC DIO and VHDCI carrier. @mntng is testing it. There is some bug/regression in the compiler/runtime: https://github.com/m-labs/artiq/issues/828 (nothing that sounds difficult to fix, though)

**Comment by hartytp** [comment]
@sbourdeauducq Thanks for confirming.

**Comment by arpitagrawal23** [comment]
Hello Everybody
I'm working with @jbqubit and have a Zotino board with me (connected via KC705 and VHDCI). I'm aiming to contribute to writing driver for Zotino. If any of you have some code in your git branch, would you let me know please? I can use it to avoid duplication of efforts and build something that is not yet done. Thank you.

**Comment by sbourdeauducq** [comment]
The driver itself is there, it just needs "glue". You should talk with @mntng, I'll write you both an email.

**Comment by dhslichter** [comment]
@gkasprow @hartytp we made a duplicate of the Zotino board filters, running from an AD5360 DAC, for some work in our lab.  Measured the output noise, which appears to be strongly influenced by environmental pickup after the filters, but with ferrites on the cable from DAC box to spectrum analyzer we measured the output noise to be ~2-3 nV/rtHz in the 1-10 MHz band, with some structure but nothing really exceeding 4 nV/rtHz.  Don't have the data traces handy, but it'd be a good sanity check to compare once @gkasprow gets the DAC testing set up.

**Comment by gkasprow** [comment]
I'll start DAC evaluation soon, once I finish tests with ADC. Hopefully this will happen this week.

**Comment by hartytp** [comment]
@gkasprow How many of these boards do you have populated atm? If you have more than one, can you send me one now, please? That way we may be able to help you with some of the testing.

**Comment by gkasprow** [comment]
@hartytp We have 3 at WUT, one in MLabs and one in Joe's hands.

**Comment by gkasprow** [comment]
@hartytp I just assembled test suite with original ADI devkit. Unfortunately it allows only DC tests, so we will need ARTIQ diver very soon.

**Comment by jordens** [comment]
AFAIK @mntng @sbourdeauducq @arpitagrawal23 @jbqubit are all working on that driver.

**Comment by hartytp** [comment]
> @hartytp We have 3 at WUT, one in MLabs and one in Joe's hands.

- [ ] @gkasprow Can you post us 1 asap then, please? That way, we can help with some of the testing.

-----

- [ ] @gkasprow @cjbe Please can you double check the list of tests above and confirm if there are any other tests you think we should do (or if there are any we shouldn't bother doing)?

> Unfortunately it allows only DC tests, so we will need ARTIQ diver very soon.

Most of the tests currently on the list can be done with a DC signal from the DAC. If you confirm the DC performance and the step response (90%-10% rise time), we can probably help with the AC measurements.

> AFAIK @mntng @sbourdeauducq @arpitagrawal23 @jbqubit are all working on that driver.

That's quite a lot of collective programming might! Any idea when this will land?

**Comment by gkasprow** [comment]
@hartytp Once I cnfirm it cooperates with devkit, will post it ASAP. Hopefully will be able to post on Monday.

**Comment by hartytp** [comment]
@gkasprow Great, thanks! Can you post some of the adapter boards as well, please (BNC->IDC and HD68 to IDC)? That will help with testing.

**Comment by gkasprow** [comment]
@hartytp sure.

**Comment by gkasprow** [comment]
@hartytp  Today I finally did tests with DAC boards. Digital part works well and I'm able to set voltage on specified output. So  @jbqubit 's student should be able to make it running.
I found some strange issue. With wrongly soldered decoupling capacitor on -12V rail, the voltage is around -2.5V and DAC gets damaged in such conditions. It happened on 2 of the boards.
This would not happen in correctly working boards with correctly installed capacitors, but I'm curious what is the mechanism of the damage.

**Comment by gkasprow** [comment]
@jbqubit please make sure that your student won't connect supply of the board with wrongly soldered caps.

**Comment by hartytp** [comment]
> @hartytp Today I finally did tests with DAC boards. Digital part works well and I'm able to set voltage on specified output. 

Great! Can you ship us a board next week then, please? Let me know when you're ready to start  analog testing.

**Comment by gkasprow** [comment]
@hartytp we will manage it. With devkit I can do only DC tests (crosstalk, stability)

**Comment by arpitagrawal23** [comment]
> @jbqubit please make sure that your student won't connect supply of the board with wrongly soldered caps.

I was not aware of this and had connected power supply several times over the last few weeks. Now, even after correcting the capacitor polarity, the board is not working. My code seems fine, I've verified SPI signals on the oscilloscope, but I'm not getting output at terminals of Zotino board. While there are chances that my code is buggy, but can we eliminate the possibility of Zotino being broken by doing some test on it?

**Comment by gkasprow** [comment]
Is your DAC chip getting hot?

**Comment by sbourdeauducq** [comment]
@gkasprow since we don't have Altium and this is not on the silkscreen: where exactly are those caps? I measured the voltage on the three tantalum caps on the top side and they are polarized correctly on the HK board.

**Comment by gkasprow** [comment]
The capacitors are mounted correctly according to PCB silkscreen. They are placed in wrong way on the schematics.
Here are pics with wrongly mounted capacitors:
![obraz](https://user-images.githubusercontent.com/4325054/31660452-c1cf6eae-b337-11e7-8e7b-8a8e9274438f.png)
![obraz](https://user-images.githubusercontent.com/4325054/31660502-e78563ec-b337-11e7-894a-d6ed64d31faf.png)


You can also use free Altium viewer.

**Comment by sbourdeauducq** [comment]
Ok, the HK board had been fixed.

**Comment by gkasprow** [comment]
A few more measurements:
![tek00063](https://user-images.githubusercontent.com/4325054/31767544-c0711518-b4cb-11e7-9b19-05cdbbcf5e71.png)
![tek00064](https://user-images.githubusercontent.com/4325054/31767547-c23a9216-b4cb-11e7-88e6-e6b23d738e0f.png)

**Comment by gkasprow** [comment]
As a next step I switched heating on and noticed that regulator disabled heating MOSFET at 50 degrees but IC kept heating itself until it failed.
On two other boards I had same situation.
So for some reason, heating of DAC chip abve 50 degrees causes some internal latchup which starts consuming too much current from -12V source until LDO overheats and disables its output.
So the DAC boards work providing that we don't heat the DAC chip. There must be a reason behind that.
Any ideas?

**Comment by jordens** [comment]
You will want to load the offset DACs with 0x2000 for all tests.

**Comment by gkasprow** [comment]
I measured what happens during power on cycle.
And it's quite interesting.
3V3 wakes up much earlier than rest of rails and is't obvious.
Then goes -12 and 12V and then reference. And this reference causes spike on negative rail. Such non-monotonic supply may cause latchup and such effect is observed. 

![tek00068](https://user-images.githubusercontent.com/4325054/31771103-1ae8ab82-b4db-11e7-8ffb-82beb3d75363.png)

**Comment by gkasprow** [comment]
there are visible oscillations and it is caused by the fact that LC1964 is missing ceramic cap at its output. There is such cap close to the DAC but it seems the equivalent ESR is too high. So I added 1uF 0805 cap and oscillations are gone. But the spike is still there...

![tek00069](https://user-images.githubusercontent.com/4325054/31771408-55354a60-b4dc-11e7-9191-fc5f1ddba20f.png)

**Comment by gkasprow** [comment]
I lowered negative power supply to -4.8V. And there is no spike so far. DAC stays cold.
![Uploading tek00070.png…]()

**Comment by hartytp** [comment]
@gkasprow What's happening on the other voltage rails? e.g. the +-13V rail and the OpAmp supply +-12V?

**Comment by gkasprow** [comment]
They are stable. Do you think they could case any issue? The output amp is conneted by 2.8k resistor to the DAC so nothing should happen.

**Comment by gkasprow** [comment]
I connected shutdown pin of VSS to reference:
![tek00070](https://user-images.githubusercontent.com/4325054/31773899-73f1289a-b4e4-11e7-807f-74a7d193cd07.png)

And this is how the opamp rail looks like.
blue - positive one:
![tek00071](https://user-images.githubusercontent.com/4325054/31773931-8ad72f28-b4e4-11e7-9a8a-726055611062.png)

And negative one (blue)
![tek00072](https://user-images.githubusercontent.com/4325054/31773943-959669a6-b4e4-11e7-812c-cf870f84390d.png)

**Comment by hartytp** [comment]
hmm...the data sheet doesn't really mention any special sequencing requirements. e.g the digital supply must turn on before we apply signals to the digital pins. But, that's about it.

**Comment by hartytp** [comment]
@gkasprow To check that I understand you, the issues you see are:

1. When the reference switches on, there is a large current draw on the DAC's negative supply, which causes that rail to sag by a couple of volts. (But no obvious signs of latch up etc).
2. When you heat the DAC above 50deg it draws a large amount of current from the negative rail until it self destructs.

**Comment by gkasprow** [comment]
@hartytp That's correct. I have a suspicion that the DAC is already on the edge of latchup which gets accelerated when it's hot. And such latchup condition may happen when there is wrong power sequence or supply rises in non monotonic way.

**Comment by gkasprow** [comment]
Another, standard condition for latchup are digital signals before power supply. But this should be the case only with DVCC rail which in our case is same as other logic.

**Comment by sbourdeauducq** [comment]
FWIW @mntng also had an episode where the DAC got very hot after powering the board. Did not happen in subsequent power-ups AFAICK.

**Comment by gkasprow** [comment]
I already damaged 3 DAC chips which are quite costly so would like to find a cause first before I burn another ones.

**Comment by gkasprow** [comment]
So I will modify the board in such way that opamps supply will go first,then VSS, VDD, DVCC as it is recommended by msamera[ here](https://ez.analog.com/message/164258).

**Comment by gkasprow** [comment]
@hartytp this will take a few days, so I can send you Novo boards first. OK?

**Comment by hartytp** [comment]
@gkasprow Yes, please do send Novo asap.

> there are visible oscillations and it is caused by the fact that LC1964 is missing ceramic cap at its output. There is such cap close to the DAC but it seems the equivalent ESR is too high. So I added 1uF 0805 cap and oscillations are gone

Did you check the supplies on Novo out of curiosity? Does this issue with the negative rail oscillating affect Novo as well? (We can check this when the boards arrive if necessary)

**Comment by gkasprow** [comment]
@hartytp I cheched supplies, anyway we would see such oscillations in the scpectrum

**Comment by gkasprow** [comment]
@hartytp tomorrow they will be on a way. @jordens wants one piece as well.

**Comment by hartytp** [comment]
@gkasprow Can you ship @jordens one of the ADC boards we paid for?

**Comment by gkasprow** [comment]
@hartytp sure.

**Comment by hartytp** [comment]
Thanks!

**Comment by hartytp** [comment]
Never mind, these have already been shipped (due for delivery on Monday). @cjbe Would you mind posting one to @jordens when they arrive?

**Comment by gkasprow** [comment]
@hartytp I know.
But this may be yet another undocumented feature. For some reason 3 chips went to silicon heaven.

**Comment by hartytp** [comment]
:(

**Comment by jordens** [comment]
IIRC there were some cases with some opamps where due to power supply sequencing they latched/had transients on their **inputs** to Vcc/Vee and thereby killed what was connected there. Don't remember where I saw that though.

**Comment by hartytp** [comment]
@gkasprow What are the DAC outputs doing during these transients?

**Comment by gkasprow** [comment]
@jordens I've heard about such case. It could happen in case of bipolar inputs. But in our case we have resistors which separate DAC outputs from opamps.

**Comment by hartytp** [comment]
@gkasprow What's the plan for getting these boards up and running?

**Comment by sbourdeauducq** [comment]
> For rest of them I will need ARTIQ test software running on KC705 or Sayma

Do you have a KC705? I'm afraid Sayma will still take some time due to the RGMII issue. KC705 code should be ready within a few days.

**Comment by gkasprow** [comment]
@sbourdeauducq I have KC705. I used it to test DIO boards

**Comment by cjbe** [comment]
@gkasprow do you have an estimate for when you will ship Zotinos to Oxford?

**Comment by sbourdeauducq** [comment]
@gkasprow OK, you can now use it with the FMC DIO 32ch, the VHDCI breakout board 1.0, and a recent ARTIQ 4.0.dev build for nist_clock (8407b2c or newer). @mntng please provide details of connections and help Greg set things up.

**Comment by hartytp** [comment]
> @jordens I've heard about such case. It could happen in case of bipolar inputs. But in our case we have resistors which separate DAC outputs from opamps.

True, this is probably not the culprit in our case. But, FWIW, the isolation resistors are made less effective by the fact that the DAC has to drive 32 of them in parallel. so the total load impedance is much lower. So, it might be worth double checking that there aren't any glitches on the DAC's output during turn on/off.

**Comment by gkasprow** [comment]
I checked, all voltages including output stage start simultaneously and monotonically. 
I added Shottky diodes at the LDO outputs and it works. Now working on temperature controller.

**Comment by hartytp** [comment]
> I checked, all voltages including output stage start simultaneously and monotonically.
I added Shottky diodes at the LDO outputs and it works. Now working on temperature controller.

So the diodes fixed this problem? That's good (although, I don't understand why from the traces you posted, since the regulators were never reverse biased)!

**Comment by gkasprow** [comment]
@hartytp I don't know what killed the DACs. I observed supply levels on already dead chips.
But since I had such issues with LDO, I added diodes to make sure it does not happen.

**Comment by hartytp** [comment]
@gkasprow Sounds like a good plan.

FWIW, I always add these diodes on dual supply designs, without waiting to do tests to see if they are needed. The diodes are very cheap, and not having them can cause some really hard to debug issues.

**Comment by gkasprow** [comment]
@hartytp I fixed the temperature controller. There is a bug so diodes show inverted logic :)
It means that diodes shown ""not too high temperature" and "not too low" so when they are both on it means it is OK. One need to add two piece of wire to fix it.
Now it takes roughly 3 mins to stabilize temperature.
I turned  it on cold, loaded 0xFFFF to the X2A register and at rougly 30 degrees and voltmeter shows: 11.289V
once it reached the temperature, the output is 11.314 when too hot LED is on and  11.307 when too cold LED is on. 
I've just received brand new 34461A DMM so will publish measurements once I manage to get them out of the multimeter.

**Comment by gkasprow** [comment]
So I think I'm redy to send you 2 pcs and leave one for further investigation.

**Comment by gkasprow** [comment]
@hartytp I observe strange behaviour. Luckily I set the current limit just above nominal so the DAC was not damaged.
It works properly whole day, but when I switched on heating, just after the regulator stabilised at 50 deg, the temperature of the DAC started growing and caused overcurrent event.
Here is the plot of the DAC output.
![obraz](https://user-images.githubusercontent.com/4325054/32555205-6d489d4e-c49c-11e7-8870-38e5f5679c12.png)

**Comment by gkasprow** [comment]
So it seems we have to limit the DAC regulator temperature to avoid such situation. No idea why hot DAC behaves like this at 50 degrees. All voltages are stable to that moment, then negative current starts growing and then power supply switches off.

**Comment by hartytp** [comment]
Odd. This shouldn't cause problems, since the junction temperature should not be much hotter than the ambient, so well below the 130C max specified.

Probably worth posting a question on the analog devices forum about this.

Can you sanity check by adding a second temperature probe as close to the DAC as possible to check that the DAC isn't very hot (e.g. if the temperature controller is actually heating the DAC up more than we mean to).

Otherwise, let's reduce the DAC temperature a bit. How about 35C?

**Comment by gkasprow** [comment]
It is even stranger.
I switched on the DAC board without heating, connected DMM to the temperature sensor and I observe continuous increase of the DAC temperature. After 2 minutes it exceeds 70 degrees. Current consumption gets increased by 100mA @12V.
So it looks like the chip was already broken once I heated it last time.
The scale is 10mV/K
![obraz](https://user-images.githubusercontent.com/4325054/32557404-58fc5dca-c4a2-11e7-9422-42b7f5f8b5a7.png)

**Comment by gkasprow** [comment]
When the chip is getting hot and I touch it by my finger the current consumption gets reduced back to the initial value.

**Comment by hartytp** [comment]
> When the chip is getting hot and I touch it by my finger the current consumption gets reduced back to the initial value.

Loose connection? Something oscillating?

**Comment by gkasprow** [comment]
No, I simply cool it down :)

**Comment by hartytp** [comment]
@gkasprow A couple of unrelated thoughts while looking at the schematic:

- The grounding on the DAC's thermal island isn't great. This could lead to worse cross-talk etc. In v1.2 do you think we can increase the grounding here (e.g. make the tabs connecting this to the main PCB a bit wider) without increasing the thermal conductivity to an unacceptable level? Even a small bit of extra ground could make a big difference IMO.
- I was having trouble tracing the layout via the PDFs, so maybe this is already done, but: it would be good to make sure that the heater return current goes directly to the EEM ground pin and without going through the ground plane any more than it has to (a separate ground trace that connects to the EEM pin would be best). This will stop the (relatively large) heater current causing changes in the ground potential seen by the DAC (which could cause changes in the DAC's output voltage). It's particularly important that the heater current doesn't connect to the ground plane on the DAC's island where the ground impedance is quite high.


Note to self: max operating temperature for the DAC is 85C. thermal impedance is 25C/W, static power dissipation (no load) is 1/4W, so expect junction temperature to be approx Ta+6C=56C, so well within safe limits. Supply currents don't change significantly with temperature according to data sheet.

**Comment by hartytp** [comment]
@gkasprow Are you sure that the temperature controller was working when you first turned it on? Could it have overshot and raised the DAC temperature to over 85C? The DAC disputes a bit of power, so is it possible that the DAC is hotter than the sensor reads (e.g. if the thermal connection between the DAC and the temperature sensor isn't great)? Can you try the heating while monitoring with an external sensor to check that we never exceed the safe limits?

**Comment by gkasprow** [comment]
The heating return current is at the temperature regulator. There is 0R resistor to make sure it does not touch the ground on the island

**Comment by gkasprow** [comment]
the sensor is soldered just under the DAC thermal pad. When I touch the DAC with my finger, I see immediate change of the sensor value so thermal contact is good.

**Comment by dtcallcock** [comment]
Are you enabling the thermal shutdown function on the DAC (bit 1 of the control register)?  Whilst it might not stop the problem occurring, you'd hope it would prevent destruction.

**Comment by hartytp** [comment]
@dtcallcock good point. It also provides a good diagnostic: turn the heater on and then read out the thermal shutdown bit. If that's triggered, we know what the issue is...

**Comment by gkasprow** [comment]
The DAC without cooling heats up itself to 36 degrees. I took the DAC board with new chip. So I suspect that when I heat it to 50 degrees, Its power consumption increases and then it heats itself until destruction.
So when it's cold in the lab it works well, when it's slightly warmer this triggers uncontrolled heating "chain reaction" that temperature controller cannot stop.
So there is one solution - instead of using heating components, use little Peltier device that can do both heating and cooling and keep temperature at let's say 20 degrees. I will modify it in such way.

**Comment by gkasprow** [comment]
This is DAC temperature in self-heating mode. Sudden drop happened when I touched DAC package with my finger so it means that heat capacity of the island is very low.
![obraz](https://user-images.githubusercontent.com/4325054/32603229-3779a890-c549-11e7-818b-906b68f07ef1.png)

**Comment by gkasprow** [comment]
Another option is to add external thermal shutdown. In this way when somebody switches the board without configuration and cooling, the DAC will sill have a chance to survive.
I tried with Pelter but it works only with airflow.
But with airflow it works also without Peltier :)
Once it exceeds 50 degrees, airflow cooling cannot stop the heating trend so we must stabilise the temperature at much lower level.

**Comment by hartytp** [comment]
> The DAC without cooling heats up itself to 36 degrees. I took the DAC board with new chip. So I suspect that when I heat it to 50 degrees, Its power consumption increases and then it heats itself until destruction.

I still don't understand this. So, without the heater, the DAC heats itself from something like 20C (room temperature) to 36C, so the temperature rise is about 15C. (Given that the DAC dissipates about 1/4W, that means that the impedance of the thermal island is about 60C/W, which is indeed very low).

From the curves in the data sheet, the DAC's power dissipation hardly changes between 20C and 80C. But, even if the DAC's power dissipation doubled, that would only add an extra 15C to the temperature, which still leaves us at something like 65C, well below the 80C that the DAC is specified to operate at. So, I can't see that the DAC's power dissipation is the cause of this, as you describe.

**Comment by hartytp** [comment]
My best guess is still that the temperature controller overshoots massively during turn-on, and that this transient is what kills the DACs -- while the DAC power dissipation doesn't really change up to 80C, I can believe it runs away beyond that. 

The temperature controller can supply ?W which, given the island's tiny thermal impedance, is enough to heat the PCB up to something like 115C. This heating will occur very rapidly given how little thermal mass the island is.

Maybe the best bet is to double the size of all those heating resistors (or even a little more than double). That should keep the DAC within the safe operating point no matter what the heater does.

**Comment by hartytp** [comment]
Greg, the temperature controller is of secondary importance for this design. So, unless decreasing the heating resistor value (or lowering the loop gain to reduce overshoot) fix this problem, my vote is to leave the temperature controller disabled for the time being and move on to the other tests.

Either way, I'm satisfied that we've reached the point where you can ship two boards to us now.

**Comment by hartytp** [comment]
Some other thoughts:

1. Given how low the thermal impedance of the thermal island is, we can probably afford to connect it to the main board better (wider/shorter tabs). This will help to give the island more thermal mass, making it less sensitive to temperature fluctuations, and (by providing more ground) will also improve signal integrity.

2. We should make sure that the final temperature controller has plenty of phase margin. The loop gain will depend a bit on ambient conditions (temperature/air flow), so we need to make sure it has acceptable phase margin over a wide range of environmental conditions. This requires a lot of extra phase margin.

3. Given that the island has low thermal impedance and the temperature controller loop isn't going to be too tight (given point 2 above) the temperature of the DAC is going to be very sensitive to air currents (cf your finger measurement above). To achieve decent stability I suspect that we will definitely need to screen air currents away from the DAC. Can we add small/cheap screening cans on both sides of the board around the thermal island. These should connect to the main PCB ground.

**Comment by gkasprow** [comment]
I modified resistor values and added external shutdown. Now when it exceeds 55degrees, DAC negative rail gets disconnected and LED starts blinking. After cooling down it starts working normally.
Temperature stabilisation loop works at 42 degrees. Below is log of temperature. Interesting moments:
0: power on
5.22 fans voltage reduced from 12V to 9V
10.45 - my colleague enters the room
12.50 - a person leaves the room
15:00 fans voltage changed to  12V

![obraz](https://user-images.githubusercontent.com/4325054/32614344-57104438-c56d-11e7-8399-3fb92dfef4d0.png)

**Comment by gkasprow** [comment]
changes for rev2:
- add full shutdown, also for positive rail. Now positive regulator does not have SHDN pin.
- increase thermal capacity of the island. At the moment it reacts too nervous to slight temperature changes in the room
- fine tune regulator. I will do it with another rev1 board but I want to send this board to @hartytp tomorrow.

**Comment by hartytp** [comment]
> @hartytp 2 boards were modified - overtherm comparator was set to 55 degrees and it shuts down the negative LDO. Tomorrow morning they will be sent to UK.

Great, thanks!

**Comment by hartytp** [comment]
I still don't understand why the DACs were breaking before. Looking at figures 15 and 16 in the datasheet shows that the DAC's power consumption does not change significantly between 20C and 85C (die temperature). The DAC's die should be approx 1/4W*25C/W=6C above ambient, so if you are only heating the thermal island to 50C (approx 56C die temperature), there really shouldn't be any problem; this is well within the specified operating range of the DAC.

So, I really can't see why the temperature controller was breaking the DACs before. Are you sure that there wasn't a temporary thermal overshoot when the temperature controller was turned on? Since the thermal mass of the island is so low, the overshoot could be quite brief and easy to miss. That's really the only explanation I can see for this, since the DAC shouldn't go into thermal run away at this temperature. Or, am I missing something?

**Comment by hartytp** [comment]
Looking at the temperature plot you posted, the temperature control loop looks quite close to oscillation (lots of ringing). As I said before, the loop gain will depend on ambient temperature/air flow/etc, so we need to make sure this loop has lots of phase margin. If it's ringing now, it could easily oscillate under different conditions.

**Comment by hartytp** [comment]
> increase thermal capacity of the island. At the moment it reacts too nervous to slight temperature changes in the room

This should help a bit. But, if we increase the coupling between the island and the main PCB too far, the power required by the temperature controller will become unacceptable.

> Another issue is that thermal regulator loop time constant depends strongly on airflow. So as @hartytp wrote shielding should help a little.

I think this will be important. Either use cheap metal screening cans, or some foam glued onto the PCBs (whichever is easier/cheaper and sufficiently robust).

**Comment by dhslichter** [comment]
What's wrong with a Peltier and putting the setpoint at ~25 C or so?

**Comment by hartytp** [comment]
@dhslichter More hassle than it's worth IMO (e.g. needs a heat sink). Easier just to heat a little and add some insulation.

**Comment by gkasprow** [comment]
@dhslichter There is nothing wrong. It has almost advantages - much better stability of the step response vs air flow. It takes roughly 100..300mW of power to keep temperature at the level of 30deg. I tested today such solution.
There are however two drawbacks:
- mechanical construction, the stack need to be installed below the shield which will act as a heatsink. So this is some additional manual work needed.
-  small (<3cmx3cm) Peltiers are usually rated at 2V, so additional DC/DC converts would be necesary.

**Comment by dhslichter** [comment]
Debatable, given the issues we are seeing here.  If we had a Peltier none of this would be occurring because we would be choosing a lower setpoint.

**Comment by hartytp** [comment]
> If we had a Peltier none of this would be occurring because we would be choosing a lower setpoint.

True.

**Comment by gkasprow** [comment]
@hartytp I plan to order tiny Peltier and use nice [chip ](https://datasheets.maximintegrated.com/en/ds/MAX1968-MAX1969.pdf)from Maxim that integrates all functions. So with little additional PCB I can try to stabilize the DAC temperature.

**Comment by gkasprow** [comment]
Another option is [this](http://www.analog.com/media/en/technical-documentation/data-sheets/ADN8831.pdf), much cheaper chip from ADI

**Comment by dhslichter** [comment]
No heat island/isolation, tie everything together with nice ground strapping.  Clear space on the underside of the board for Peltier?  Not sure what the clearances are there.  If it has to be on the top side, you can't run under a can, but you rely on having the Peltier outside the can but close to the DAC, and you make the thermal conductivity through the ground plane layers high so that you have a good thermal link.  Remember the goal should just be stabilizing temperatures against noise/drifts, probably accepting a dc temperature offset which might vary from rack to rack.

**Comment by gkasprow** [comment]
@dhslichter What about placing DAC on bottom side of the PCB, then on top goes Peltier, then the heatsink protrudes through the shielding can which can be much lower than now.
I'd like to isolate the DAC from other ICs, mainly output drivers that may generate some heat when loaded. And since QFP package conducts the heat mainly through bottom thermal pad, and we want to stabilize temp of the DAC, such solution seem to be optimal.
That was my intention of this island to isolate the DAC from neighbouring opamps. I did not take into account that its thermal capacity is so low and that regulator time constant varies a lot with airflow and it might be difficult to provide such phase margin that it would work reliably in all conditions.
I noticed today when I was playing with the board that regulator works quite well with window open, it reaches temperature without oscillations. But when I closed the window, it started oscillating itself. The board is placed just about the window and outside is really cold.

**Comment by hartytp** [comment]
> I noticed today when I was playing with the board that regulator works quite well with window open, it reaches temperature without oscillations. But when I closed the window, it started oscillating itself. The board is placed just about the window and outside is really cold.

Well, that explains the mystery of the spontaneously exploding DACs ;)

**Comment by hartytp** [comment]
> through the shielding can which can be much lower than now.

Do we even need the screening can?

My vote is this: let's forget about the temperature controller for the time being and focus on testing the rest of the board. One test we plan to do anyway is the EMI sensitivity test. If we find that the DAC doesn't need screening to protect from adjacent noise sources (Kasli etc) then we can scrap this screening can, which makes the peltier mechanics a bit easier.

**Comment by gkasprow** [comment]
@hartytp the oscillations were less than 1K. Initial regulator components gave very short time constant so the heater wouldn't be able to kill DAC. I had to extend regulator time constant 20 times to make the regulator somehow stable. 
Overheating could have happened if the regulator was too slow.
Anyway I will investigate it with original component values.
The same situation happened with 3 boards so at least it is consistent :)
So first of all, what needs to be measured is step response/ bode plot of the thermal circuit in different airflow conditions. Then we can check if it is possible to choose PID values that would make such regulator stable. If not, then Peltier would be the only way to go.

**Comment by dhslichter** [comment]
One thing to think about with Peltier is that we will probably want to run in analog mode, rather than PWM mode.  There is an efficiency tradeoff, but it's probably not good to have ~amp-level pulsed currents directly next to your DAC, in terms of noise pickup :)

**Comment by dhslichter** [comment]
DAC on bottom of PCB is fine with me.  I don't think the DAC needs a shielding can necessarily.  Also, I forget how cluttered the inner layers of the board are, but if you ran the signals on inner layers with grounds on the outside, it would provide a nice "shielding can" for pickup in the long traces between DAC and amplifiers/filters, which I think is our main potential issue in terms of pickup.

**Comment by gkasprow** [comment]
@dhslichter good idea. But anyway we need to generate low voltage, high current supply for Peltier. In my case I needed 1A and 0.2V to keep DAC at 30 degrees.

**Comment by hartytp** [comment]
Reminding myself how good the temperature stability needs to be...

- Target is 10ppm drifts max, ideally more like 3ppm
- DAC stability (gain + offset): 5ppm/K
- reference stability: +-2ppm/K max
- So we'd ideally like these components stabilised to better than +-1K (0.25K would be great). That's easily achievable in good lab conditions, but probably not in a rack with DDSs etc. So, stabilisation isn't an absolute requirement, but nice to have.

OpAmps (not actively stabilised):
- offset voltage: +-3uV/K max = +-0.15ppm/K max. For +-3ppm, need temperature stability better than +-20K. (!)
- bias current: ....
- self heating: worst-cast power dissipation is driving 10/2=5V into the 500R=100mW. Thermal impedance is 50K/W (junction to PCB, this ignores other sources of heat dissipation), so that's 5K worst-case self-heating.

NB Ignoring temp co of output resistor since I assume the load is essentially capacitive.

**Comment by hartytp** [comment]
Anyway, pelter sounds good if not too expensive, but it'd be great to see the other tests done before we spend too much time on the temperature controller!

**Comment by dhslichter** [comment]
@hartytp agreed on this count, but we have to figure out a way to keep the DACs from cooking themselves into oblivion before we can do the tests :)  

The output resistor will matter if one is making ac waveforms, though.

**Comment by gkasprow** [comment]
@dhslichter I implemented simple protection mechanism that switches the DAC rail off when too hot.
One can modify other boards as well or use without heater as they are now.

**Comment by hartytp** [comment]
> The output resistor will matter if one is making ac waveforms, though.

Not necessarily: for the DC one wants ppm stability, but in most cases 0.1% drifts in the AC amplitude aren't a problem. In any case, the AC drift will likely be dominated by the caps, not the Rs.

@gkasprow Given that these resistors can take as much as 100mW (DAC short circuited) is it worth paralleling two resistors for the outputs to improve thermal performance?

**Comment by gkasprow** [comment]
@hartytp sure, we can add, there is plenty of space.

**Comment by hartytp** [comment]
@cjbe @klickverbot If you have time, do you think you can help out with some of the above tests? Particularly the ones requiring Artiq, since @gkasprow doesn't have Artiq up and running atm. Anything you can do will help speed things along (I'm keen to get onto the ADC v2 design so M-Labs can finish the noise eater).

**Comment by gkasprow** [comment]
Today I repaired and modified the DAC board I have. I left original regulator compensation components. Only temperature set point is different - 40 degrees instead of 50 degrees.
Below is temperature log. After 3 mins I switched on the fans.
![obraz](https://user-images.githubusercontent.com/4325054/32795514-32015f4e-c96c-11e7-97f7-75dd60717543.png)
 So now I can play with optimisation before Urukul arrives. Technosystem promised they will be ready on Friday.

**Comment by jordens** [comment]
Then I take it that Kasli has been handed of to Technosystem as well?
IMHO doing the redesign for Novo with the new ADC would fit the timeline nicely right now. No?

**Comment by gkasprow** [comment]
Yes, Kasli is in production right now.

**Comment by hartytp** [comment]
> IMHO doing the redesign for Novo with the new ADC would fit the timeline nicely right now. No?

If there is time to make significant progress on Novo v2.0 before Friday then I'd agree: that should take priority over further testing of Zotino. Once that is done, we have a prototype of all EEMs currently planned, so users can start testing and M-Labs can write software.

**Comment by hartytp** [comment]
> Yes, Kasli is in production right now.

Fantastic!

**Comment by jordens** [comment]
Great.
@marmeladapk is v1.0rc3 == v1.0? Could you do the release process as described?
@gkasprow Do you need input on [Novo v2.0](https://github.com/m-labs/sinara/milestone/15)?

**Comment by gkasprow** [comment]
I played a little with RC filter values.
Now the components are: Rint = 200k, Cint = 47n II (10uF+10k)
![obraz](https://user-images.githubusercontent.com/4325054/32862652-176c80ee-ca59-11e7-8804-62d370eef085.png)

**Comment by hartytp** [comment]
@gkasprow I really worry about making this temperature stabilization circuit overly complex, as the stability of the DAC is almost good enough to get away without the stabilization circuit anyway.

Let's put this issue aside for now and come back to it once we've finished testing the rest of the DAC's functionality (and designing the v2 ADC).

**Comment by hartytp** [comment]
@gkasprow Here are my thoughts about the temperature controller:

- It was a nice idea to have a really simple fixed-gain temperature controller like the one we currently have. However, as we've seen, this is unlikely to work given the potential range of ambient temperature/air flow. As a result, v1.2 will not feature this kind of temperature controller. As a result, we should not spend more time tuning this temperature controller.

- The current design for the thermal island creates problems, since the thermal mass is so low it make the DAC much more sensitive to ambient temperature fluctuations. As a result, v1.2 should either feature no thermal island (DAC directly on the main PCB) or a redesigned island with more thermal conductivity to the main PCB.

- To meet our stability target (+-10ppm) the board must be stable to +-2K. 

- For v1.2 we have two options. Either scrap the thermal island and leave it up to the users to ensure that the board temperature is sufficiently stable for their needs. Or, add a more fancy temperature controller, probably featuring a microprocessor and a peltier. If we go for the second option, I would prefer the temperature controller to be on a separate PCB that mounts on top of the DAC/thermal island. That way users who don't care about stability can ignore it. In this way, we can first design the DAC board without the temperature controller board, and design the temperature controller board later on when needed.

- In any case, since the temperature controller will need redesigning from scratch, there is no point doing any more tests on it in this version of the design.

**Comment by hartytp** [comment]
@gkasprow A couple of questions about power supplies/sequencing on the DAC:

- Why are there two separate +-12V supplies? Can't the DAC use the sample supplies as the OpAmps?
- Is it a problem if the DAC supplies turns on (off) before (after) the OpAmp supply? Would it make sense to put diodes between these supplies to ensure that the DAC can't drive the OpAmps beyond their supply rails (e.g. diode between DAC +12V and OpAmp +12V)?

**Comment by gkasprow** [comment]
@hartytp well, it is purely historical. Initial idea assumed different supply levels for DAC and amplifiers.Then we decided to have 10V range instead of 12 due to the price of precise resistors so I only modified the LDO resistors. One can also make DC/DC voltage lower 13V would be enough.
Diodes are fine.

**Comment by hartytp** [comment]
> @hartytp well, it is purely historical. Initial idea assumed different supply levels for DAC and amplifiers.Then we decided to have 10V range instead of 12 due to the price of precise resistors so I only modified the LDO resistors. One can also make DC/DC voltage lower 13V would be enough.

Okay, in the next revision let's scrap 1 set of +-12V regulators and reduce the SMPS outputs. I've added a note to the top post.

**Comment by cjbe** [comment]
@gkasprow I am trying to bring up a Zotino. When I power it up and program it I get the correct voltages out of the DAC. However after a minute the DAC outputs start dropping out to 0V for ~500ms every 2s. This appears to be due to the DAC_VSS supply cutting out periodically. The input to IC33 (the DAC_VSS reg) does not drop out.

When the supply drops out the 'Lo' LED comes on, and when the supply is good the 'OK' LED comes on. The 'Htr' LED is permanently on. However the heater jumper (SW1) is not present, hence the heater should not be doing anything? Also the P12V0 supply for the heater circuit is steady, so this correlation is not due to a common power rail dropping out.

Any ideas?

**Comment by cjbe** [comment]
Aha - the hacked on over-temperature cutoff was activating. Even without the heater, the DAC temperature rose above 50 C in a warm lab with little airflow. Adding a fan keeps the DAC around 40C.

**Comment by jordens** [comment]
It would be interesting to benchmark the thermal simulations on Zotino. What's the prediction for the DAC temperature on the island at its idle power with no/little air flow?

**Comment by hartytp** [comment]
> Even without the heater, the DAC temperature rose above 50 C in a warm lab with little airflow. Adding a fan keeps the DAC around 40C.

Ouch! That's not good.

We really need to fix this soon for the v1.1 boards (next revision), as the current design makes Zotino pretty hard to use.

My suggestion is this:

1. Increase the thermal conductivity between the thermal island and the main board by at least 1 order of magnitude. When idling without active temperature control, the DAC should only be a couple of degrees above the rest of the board (this is important, as these boards should be designed to function properly even without active temperature control). Once this is done, temperature controlling the DAC will also control the main PCB to a certain extent. AFAICT, that should be fine if we use a peltier.

2. Remove the temperature controller circuitry from the PCB -- this kind of simple temperature controller is unlikely to work given the wide range of ambient conditions that it has to operate in.

3. Also remove the DAC's thermal shutdown, this should not be necessary once we improve the design of the thermal island

4. Keep the temperature sensor on the thermal island, and bring it to a header which can connect to a separate temperature controller PCB. This header should also supply +12V power for such a board. Could also connect I2C to this header if we want to provide control interface for the temperature controller.

5. Add mounting holes near the thermal island that could be used to mount a separate temperature controller PCB + peltier.

6. At some point in the future (not a high priority at the moment) we can design a separate temperature controller PCB. The stackup is: DAC, peltier, temperature control PCB. This PCB has a heat sink, peltier driver and microprocessor. Are we okay mounting the peltier onto the DAC, or would it be better to mount it directly onto exposed copper on the thermal island?

@gkasprow @cjbe @jordens Are you happy with that plan?

**Comment by jordens** [comment]
Yes. Right now I'd maximize passive temperature stability by minimizing thermal impedances and maximizing thermal mass (opposite as of now). The thermal impedance between island and rest of the board could even be made a placement option during fabrication. Later, and with a peltier we can stabilize DAC temp around its average. There are already plenty of commercial temperature controller PCBs around. Add a mounting hole or two for the controller PCB and a header for sensor, I2C, and 12V.

**Comment by jbqubit** [comment]
Is the current front panel width enough to accommodate the added thickness from the separate temperature controller stack?

**Comment by hartytp** [comment]
> Is the current front panel width enough to accommodate the added thickness from the separate temperature controller stack?

If not, one can easily buy a separate filler panel, so this isn't something we have to worry about for this design IMO.

**Comment by gkasprow** [comment]
For use cases that need special temperature control, we can install general purpose temperature controller in neighbouring slot. I mean one that will be developed as a part of [planned hardware.](https://github.com/m-labs/sinara/wiki/PlannedHardware)

**Comment by gkasprow** [comment]
And for this purpose we may simply add some connector that would give access to the temperature sensor and Peltier solder pads. The connector can be located close to EEM and we could use 6 pin IDC. The same one would be placed on the temperature controller board

**Comment by hartytp** [comment]
> And for this purpose we may simply add some connector that would give access to the temperature sensor and Peltier solder pads. The connector can be located close to EEM and we could use 6 pin IDC. The same one would be placed on the temperature controller board

> For use cases that need special temperature control, we can install general purpose temperature controller in neighbouring slot. I mean one that will be developed as a part of planned hardware.

Sounds good!

**Comment by hartytp** [comment]
Top post updated with new plan.

**Comment by cjbe** [comment]
@gkasprow I am having a few issues with the Zotino prototype you sent me:
* DAC channel 7 has stopped working. It was working when I received the board, and the board has not been through any trauma. The failure appears to be in the DAC chip itself (output is ~12R to ground)
* There is some failure mode of the board that sometimes causes all of the DAC outputs to drop to zero briefly. Once this failure is triggered, the outputs cycle between 0 and correct once every few seconds. After a while, or after power cycling, this stops. I have not been able to reproduce this properly with instrumentation attached.

Also a few other minor niggles:
* The op-amp on channel 1 was dead on the board I received - I replaced it, and have not seen any more  die.
* J5 (one of the DAC output IDCs) was populated the wrong way around.

**Comment by jordens** [comment]
@cjbe How are you handling the thermal issues now? Turning of the temp control, adding a heat sink on the DAC or with just the fan?

**Comment by gkasprow** [comment]
@cjbe The boards I modified could have some trauma before I discovered the mechanism that killed the DAC chips. So it's quite possible that your DAC was already slightly broken which explains why it self-triggered the thermal protection circuit which does not happen before on boards without heater enabled.
So I'd suggest replacing the DAC chip - it's easy providing that you have hot air gun and the flux. If you have troubles with it, ship it to me and I will replace it.

**Comment by cjbe** [comment]
@jordens the Zotino is sitting next to an A/C vent in an open frame rack, and has a fan pointing at it. The thermal management on the Zotino is disabled, but the DAC internal thermal cutoff is enabled, but has not ever tripped (it trips at a junction temperature of ~120C!).

**Comment by cjbe** [comment]
@gkasprow I already have one on order - I will let you know if I see any funnies after replacing the DAC.

**Comment by hartytp** [comment]
@gkasprow (or @cjbe if you have time once your new DAC arrives): I think there are enough issues with this version of Zotino that it has become so hacked/fragile that further testing is hard. So, I propose that we do a couple of very quick tests to confirm basic performance and then move on to v1.2 fixing all known issues.

Please could you make the following measurements:
- 10% to 90% rise time for a full-scale signal
- Noise spectrum (nV/rtHz) for 0V and +-10V signals. Ideally, use an instrument with a very low noise floor for this (1nV/rtHz or so) and measure in the bandwidth of something like 0.1Hz to 1MHz.

If those measurements look okay then I'm happy moving on to v1.2

**Comment by hartytp** [comment]
@cjbe how did the new DAC work out? Any luck?

@gkasprow I know there's a lot on at the moment, but is there any chance you could do those two tests (rise-time and noise spectrum) next week? In it's current state, Zotino doesn't really work well enough for prototype testing/evaluation, so I'd really like to try to send v1.1 (the next revision) off for manufacture before Christmas if at all possible.

Thanks!

**Comment by gkasprow** [comment]
@hartytp some rise-time measurements were already [done](https://github.com/m-labs/sinara/issues/227#issuecomment-337871416) scroll the page up.
I have a setup done so will do more measurements once ship Urukul 9912.

**Comment by hartytp** [comment]
Thanks for the reminder @gkasprow 

Looking at rise-time measurements in detail:

For the main slope: dV=17.6V, dt=25us which is broadly consistent with a slew rate of approx 0.7V/us. That's about what we expect: DAC slew rate is 1V/us, OpAmp is 20V/us. 

This is a bit slow compared with @dhslichter's SPICE simulations (see plot on schematic). Presumably that is because the step response plot is only for the filter, and doesn't include the DAC's slew-rate limit? (@dhslichter can you confirm this?)

Primary overshoot looks to be about 120mV and settles out in a bit under 10us. Looks like our circuit is a little more damped than the situation (120mV versus something like 180mV in the sim).

Overall, those rise-time measurements look fine. 

Point of the currents tests is just to make sure that this design more-or-less meets specification; we will make more careful measurements on v1.1 (with bug fixes). From that measurement, it looks like the board does behave properly from a rise-time perspective.

Once we check the noise performance, I'm happy to move onto v1.1

**Comment by hartytp** [comment]
Answering my own questions: yes, AFAICT, the simulation does only include the filter (see https://github.com/m-labs/sinara/issues/185). This explains the slight differences between filter and measured performance. NB we should either remove this simulation or document what it means more clearly!

**Comment by dhslichter** [comment]
Yes, it's just the filter.

**Comment by cjbe** [comment]
@gkasprow I replaced the DAC on my Zotino. This fixed all of the output channels.

After an hour of operation the DAC chip remained cool (~30C). However after leaving the Zotino powered overnight the DAC chip was very hot (~90C). 
After turning the power off and letting it cool down, then reconnecting it, the DAC chips gets very hot in a few minutes.

This seems to suggest something very alarming is going on. @gkasprow : any suggestions?

**Comment by cjbe** [comment]
I made some noise measurements of the prototype Zotino. Note these are with a DAC chip that is behaving oddly (getting very hot), but I think all the interesting artifacts are not related to the DAC:

![noise](https://user-images.githubusercontent.com/13227288/34040998-13b5040c-e18e-11e7-8645-bf42baef6de7.png)

The noise floor is about 350 nV/rtHz. The pickup with fundamental 44.5 kHz is from the switching power supply. The amplitude of the fundamental is 70 uV rms.

I measured with the DAC output at 0V or 10V. The Zotino was connected to the test breakout board with 20cm of ribbon cable, then to a floating input of a Rhode & Schwarz UPV audio analyzer.

**Comment by gkasprow** [comment]
@cjbe did you disable my overtherm protection circuit? 
Heating was also disabled?
Cooling was on all the time?

**Comment by gkasprow** [comment]
@cjbe The DC/DC convertes run at much higher frequency, so I'm not sure if this is a source.

**Comment by cjbe** [comment]
@gkasprow I believe it is the DC/DC, as I looked at the -Vout pin of IC1 and saw ~10mV of ripple at 44.5 kHz.

**Comment by cjbe** [comment]
@gkasprow I disabled your over-temperature circuit (it is still present, but I stopped it disabling the DAC supply). I have enabled the DAC internal over-temperature protection, which triggers at 120C, but so far this has never triggered.

I have the card sitting on my desk in a ~25C room - there is no active cooling. As it was happy for an hour, I assumed that there was no possibility of thermal runaway.

**Comment by gkasprow** [comment]
@cjbe it happened to my board that after some time, when the temperature or the air flow in the room changed, it got its thermal runaway.

**Comment by gkasprow** [comment]
And these 44.5kHz are surprisingly low . I'd expect at least 200kHz judging the converter size and its power.
I'll look deeper to these ripples.

**Comment by a-shafir** [comment]
I have not found CC6-1212DF-E base freq but by the emi testing docs it looks like ~600KHz. TPS62175DQC is around 1MHz

44.5kHz and it's harmonics can be some interference from the D-class audio around (via ground loops etc). 
Or it can be noise from the digital circuits, the controller (KC705?) etc.

**Comment by gkasprow** [comment]
@cjbe could you connect just power supply to the DDS board, no interference and measure it one again?
Or the best would be to operate it from 12V battery to make sure nothing gets via the supply lines.
Shat wall wart do you use? Some of them have very poor EMC performance

**Comment by cjbe** [comment]
@gkasprow I am now using a nice linear power supply to power the Zotino.
This plot shows both the voltage at the Zotino power supply input (Net P12V0), and directly at the output pin of the switchmode (-Vout):

![ripple](https://user-images.githubusercontent.com/13227288/34050301-4c86885e-e1b2-11e7-8a3e-cf5285e0c0ab.png)

So it looks like the ripple is not coming from the power supply to the zotino, but is generated on the Zotino. It looks like the switcher frequency is ~600 kHz, but there is ripple at ~41 kHz

**Comment by cjbe** [comment]
A significant part of this issue seems to be due to the DAC overheating:

After directly after the Zotino is turned on it draws ~290 mA at 12V, and the modulation on the output at 41 kHz is at ~-110 dBVrms. Over a few minutes the current creeps up by a few tens of mA, then fairly rapidly climbs to 500 mA, then wanders between 500 and 640 mA over tens of seconds. As this happens, the modulation peak increase to ~-80 dBVrms.

**Comment by gkasprow** [comment]
@cjbe your DAC is dead, negative LDO overheated and thermal shutdown gets enabled.
This looks like the DC/DC converter gets unstable and its internal feedback loop causes oscillations at 41kHz

**Comment by hartytp** [comment]
@cjbe What's happening to the reference voltage during this period?

Thinking outloud here, but:
- Let's assume that when the DAC says its max operating temperature is 120C it really means that it can take this CW while still meeting specification. Moreover, let's assume that your tests do show that the DAC die never exceeds 120C. As a result, the DAC overheating isn't the problem.
- The reference is only rated to +70C. Given that it's also on the thermal island I can well believe that it can exceed this.
- The DAC can only take 5V5 reference input, so if the reference fails and applies DAC_VDD (+12V) to this pin, that will probably kill the DAC.

**Comment by a-shafir** [comment]
it is not impossible to have parasitic oscillation process between the switching regulators and the linerar regulators etc.
There are some instructors in the power circuit etc.
The regulators control loop b/w possible is not enough to suppress it fully.

The capacitors used on the board (22uF etc) are relatively small considering the load currents.
Also the power filtering instructors can be close to saturation and increase the stability problem.

@cjbe can you try to add a big (over 100uF) tantalum capacitor parallel to C136, C130, C131, C141, C133 etc (can be one by one)?

If this is related to the oscillation like i suspect shall at lest change the ripples so it will indicate the problem.

**Comment by hartytp** [comment]
@a-shafir No, this is just the SMPS in overcurrent mode. It's only a 0.25A model, and once the DAC fails we draw a hell of a lot more than that.

Adding large caps probably will change the frequency of this oscillation, but that doesn't tell one much (it would be true for many different noise sources).

**Comment by hartytp** [comment]
@cjbe Thanks for the noise measurement! Pretty sure that's not actually the noise floor of the circuit when it's working properly! If the SMPS is in overcurrent mode then this data is all suspect. :(

**Comment by gkasprow** [comment]
be careful with adding too many capacitors at the SMPS output - it can easily became unstable.
Inductors have Isat = 950mA, so there is no way to saturate them.
Ripples at the output of SMPS are something normal, I simply didn't expect 44kHz while designing the LC filter

**Comment by gkasprow** [comment]
Almost the same power supply is used on ADC board and  I get the noise floor at -120dB easily.

**Comment by hartytp** [comment]
> I simply didn't expect 44kHz while designing the LC filter
> Almost the same power supply is used on ADC board and I get the noise floor at -120dB easily.

Right! And, my guess is that once we stop killing the DACs this ripple will go away. It's almost certainly a symptom of DAC death rather than a problem in itself.

**Comment by hartytp** [comment]
@gkasprow @cjbe At this point, I think we should just stop testing this version of Zotino! Replacing those DACs is too time consuming and expensive! Whatever the exact failure mechanism is, it's most likely related to the thermal island -- and, even if it's not, I think we need to eliminate the island as a potential source of problems.

Can we just fix the known issues (see top post of this thread) and order a couple more boards?

**Comment by hartytp** [comment]
Let's do this as soon as reasonably possible, since I really want to get a version of this PCB that we can actually test.

**Comment by hartytp** [comment]
@cjbe @gkasprow One thing though:

If either of you have a chance, it would be interesting to measure the noise spectrum of the filter with the DAC removed from the board (just ground the DAC end of R24).

This at least lets us check the filter's noise. Hopefully with the DAC gone, the SMPS will also start behaving.

**Comment by gkasprow** [comment]
Tom, let mi test Kasli first. Just after that I will focus on DAC board.

**Comment by hartytp** [comment]
@gkasprow Sounds like a good plan.

**Comment by cjbe** [comment]
@gkasprow I just monitored the reference output as the Zotino overheated from cold. The reference output stayed at 5.00V even after everything has reached thermal equilibrium. 

For reference, the temperature sensor was reading 27 degrees with the board just turned on from cold, and ~85-88 degrees when the board was at thermal equlibrium.

**Comment by gkasprow** [comment]
@cjbe thermal equilibrium means overheated negative regulator :) You can check value of negative rail in such condition.

**Comment by cjbe** [comment]
@gkasprow I check the DAC supplies in the overheated condition, both +12V and -12V are nominal.

**Comment by hartytp** [comment]
So, AFAICT, the absolute max operating temperature of the DAC is +85C, but the overtempeature cutout only trips at 120C...

**Comment by gkasprow** [comment]
so it seems that my DACs died in more spectacular way...

**Comment by cjbe** [comment]
@hartytp  @gkasprow I have measured the noise of the Zotino with the filter input grounded, and the DAC removed (to not overload the SMPS). The noise curve is below (previous DAC at 0V data for reference):
![grounded noise](https://user-images.githubusercontent.com/13227288/34679767-dc66e54c-f48e-11e7-98f0-92356c4f85bb.png)
There still seems to be significant noise at ~100 kHz, even without the dying DAC heavily loading the SMPS

**Comment by jordens** [comment]
The low frequency stuff is mysterious. Shouldn't that SMPS be at ~1 MHz >> 50 kHz?

**Comment by hartytp** [comment]
Thanks for doing that Chris!

There's obviously something severely broken with this board. IMO there is no point debugging this further, given how much abuse these boards have been through. Better to move onto producing the v1.1 design asap.

**Comment by gkasprow** [comment]
@cjbe could you short the output CMC and repeat the measurement?
It can pickup some external noise...

**Comment by cjbe** [comment]
@gkasprow I assume you mean short both coils of the CMC in order to effectively remove it? I am testing this with your 'ribbon to BNC' test boards, so there is both a CMC on the Zotino, and a CMC on the test board. I assume I should short both?

**Comment by gkasprow** [comment]
Yes.

**Comment by hartytp** [comment]
@gkasprow The broad-band noise also looks orders of magnitude higher than it should be, so I don't think this is the chokes (we always blame those poor things and they've never been the problem yet!)

Can we defer further tests until we've got the v1.1 boards ready, please?

**Comment by cjbe** [comment]
@gkasprow I shorted out the CMC's on a grounded channel - this did not significantly change the ~100 kHz noise, but increased the 50 Hz and harmonics pickup by >50 dB (so the CMC's are doing a good job!)

**Comment by hartytp** [comment]
Looks like (at least part of the problem) is the power supply lines.

If you want to try to track this down further, you could try replacing the SMPS with a bench PSU and see what happens to the noise. Also, that would let you check the current draw on the +-13V lines and check that they match the power budget...

**Comment by gkasprow** [comment]
I'd suggest one crazy thing -  measurement without the power supply.
@cjbe What tools are you using to measure the noise density? 
I'd like to recreate it in my lab.

**Comment by gkasprow** [comment]
@hartytp I updated schematics of Zotino and initial components placement

**Comment by cjbe** [comment]
@gkasprow I am using an R&S UPV Audio Analyzer.
I will do some more sanity checks tomorrow to trying and narrow down the source of the noise.

**Comment by gkasprow** [comment]
I have access to Agilent SSA, in dedicated faraday cage room. I will check tomorrow what is the lowest frequency it can measure.

**Comment by hartytp** [comment]
Closing this as we're done with tests for Zotino v1.0. v1.1 discussion at #446
